acceptanceTests/tap_test.go

```diff
@@ -973,3 +973,197 @@ func TestTapDumpLogs(t *testing.T) {
+
+func TestTapDaemon(t *testing.T) {
+	if testing.Short() {
+		t.Skip("ignored acceptance test")
+	}
+
+	tests := []int{50}
+
+	for _, entriesCount := range tests {
+		t.Run(fmt.Sprintf("%d", entriesCount), func(t *testing.T) {
+			cliPath, cliPathErr := getCliPath()
+			if cliPathErr != nil {
+				t.Errorf("failed to get cli path, err: %v", cliPathErr)
+				return
+			}
+
+			tapDaemonCmdArgs := getDefaultTapCommandArgsWithDaemonMode()
+
+			tapNamespace := getDefaultTapNamespace()
+			tapDaemonCmdArgs = append(tapDaemonCmdArgs, tapNamespace...)
+
+			tapCmd := exec.Command(cliPath, tapDaemonCmdArgs...)
+			t.Logf("running command: %v", tapCmd.String())
+
+			viewCmd := exec.Command(cliPath, getDefaultViewCommandArgs()...)
+
+			t.Cleanup(func() {
+				if err := runMizuClean(); err != nil {
+					t.Logf("error running mizu clean: %v", err)
+				}
+				if err := cleanupCommand(viewCmd); err != nil {
+					t.Logf("failed to cleanup view command, err: %v", err)
+				}
+			})
+
+			if err := tapCmd.Run(); err != nil {
+				t.Errorf("error occured while running the tap command, err: %v", err)
+				return
+			}
+
+			t.Logf("running command: %v", viewCmd.String())
+			if err := viewCmd.Start(); err != nil {
+				t.Errorf("error occured while running the view command, err: %v", err)
+				return
+			}
+
+			apiServerUrl := getApiServerUrl(defaultApiServerPort)
+
+			if err := waitTapPodsReady(apiServerUrl); err != nil {
+				t.Errorf("failed to start tap pods on time, err: %v", err)
+				return
+			}
+
+			proxyUrl := getProxyUrl(defaultNamespaceName, defaultServiceName)
+			for i := 0; i < entriesCount; i++ {
+				if _, requestErr := executeHttpGetRequest(fmt.Sprintf("%v/get", proxyUrl)); requestErr != nil {
+					t.Errorf("failed to send proxy request, err: %v", requestErr)
+					return
+				}
+			}
+
+			entriesCheckFunc := func() error {
+				timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+
+				entriesUrl := fmt.Sprintf("%v/entries?limit=%v&operator=lt&timestamp=%v", apiServerUrl, entriesCount, timestamp)
+				requestResult, requestErr := executeHttpGetRequest(entriesUrl)
+				if requestErr != nil {
+					return fmt.Errorf("failed to get entries, err: %v", requestErr)
+				}
+
+				entries := requestResult.([]interface{})
+				if len(entries) == 0 {
+					return fmt.Errorf("unexpected entries result - Expected more than 0 entries")
+				}
+
+				entry :=  entries[0].(map[string]interface{})
+
+				entryUrl := fmt.Sprintf("%v/entries/%v", apiServerUrl, entry["id"])
+				requestResult, requestErr = executeHttpGetRequest(entryUrl)
+				if requestErr != nil {
+					return fmt.Errorf("failed to get entry, err: %v", requestErr)
+				}
+
+				if requestResult == nil {
+					return fmt.Errorf("unexpected nil entry result")
+				}
+
+				return nil
+			}
+			if err := retriesExecute(shortRetriesCount, entriesCheckFunc); err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+		})
+	}
+}
+
+func TestTapDaemonMultipleNamespaces(t *testing.T) {
```

Comment:
Also needs a more detailed name.
From just looking at the name of this function, I thought that it tests the same thing as `TestTapDaemon` only in the specified namespaces, but it doesn't.
Maybe `TestDaemonSeePods`?

Reply:
Add a negative test (in this function or a different one) that tests that a pod in another namespace is not seen. (`unexpectedPods`)